mdal/frmts/mdal_selafin.cpp

```diff
@@ -49,7 +49,22 @@ void MDAL::SelafinFile::initialize()
   mFileSize = mIn.tellg();
   mIn.seekg( 0, mIn.beg );
 
-  mIsNativeLittleEndian = MDAL::isNativeLittleEndian();
+  mChangeEndianness = MDAL::isNativeLittleEndian();
+
+  //Check if need to change the endianness
+  // read first size_t that has to be 80
+  size_t firstInt = readSizet();
+  mIn.seekg( 0, mIn.beg );
+  if ( firstInt != 80 )
+  {
+    mChangeEndianness = !mChangeEndianness;
+    //Retry
+    firstInt = readSizet();
+    if ( firstInt != 80 )
+      throw MDAL::Error( MDAL_Status::Err_UnknownFormat, "File " + mFileName + " could not be open" );
+    mIn.seekg( 0, mIn.beg );
+  }
```

Comment:
can you please move all this code (lines 52-66) to a new local function `void determineEndianness()`

Reply:
not sure it is a good idea and I don't see good reason because this code is used only from here. This code has to be executed from the beginning of the file and will change the position in the file if it is called from else where. In the initialize() method, no risk of an unwanted position change in the file.